pywt/_multidim.py

```diff
@@ -147,9 +151,43 @@ def dwtn(data, wavelet, mode='symmetric'):
     return dict(coeffs)
 
 
+def _fix_coeffs(coeffs):
+    missing_keys = [k for k, v in coeffs.items() if
+                    v is None]
+    if missing_keys:
+        raise ValueError(
+            "The following detail coefficients were set to None: "
+            "{}.".format(missing_keys))
+
+    invalid_keys = [k for k, v in coeffs.items() if
+                    not set(k) <= set('ad')]
+    if invalid_keys:
+        raise ValueError(
+            "The following invalid keys were found in the detail "
+            "coefficient dictionary: {}.".format(invalid_keys))
+
+    key_lengths = [len(k) for k in coeffs.keys()]
+    if len(np.unique(key_lengths)) > 1:
+        raise ValueError(
+            "All detail coefficient names must have equal length.")
+
+    # make sure the entries of coeffs are all numpy arrays
+    coeffs = dict((k, np.asarray(v)) for k, v in coeffs.items())
+
+    ndim = np.unique(key_lengths)
```

Comment:
This returns an array of length one (we've already checked above to make sure there is only one unique value). I'd check in